Fix options switch alignment within settings

jarolrod opened the pull request:
Fixes alignment of the option switch within a settings row.

| master | PR |
| --------- | ---- |
| ![master-option](https://user-images.githubusercontent.com/23396902/184701651-89911f6c-5c9c-444e-8d37-93c7ba22e41b.png) | ![fix-option-pr](https://user-images.githubusercontent.com/23396902/184701668-44b61954-abe2-4af4-9c4e-76a7690ef466.png) |


Closes bitcoin-core/gui-qml#159

[![Windows](https://img.shields.io/badge/OS-Windows-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/win64/insecure_win_gui.zip?branch=pull/161)
[![Intel macOS](https://img.shields.io/badge/OS-Intel%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos/insecure_mac_gui.zip?branch=pull/161)
[![Apple Silicon macOS](https://img.shields.io/badge/OS-Apple%20Silicon%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos_arm64/insecure_mac_arm64_gui.zip?branch=pull/161)
[![ARM64 Android](https://img.shields.io/badge/OS-Android-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/insecure_android_apk.zip?branch=pull/161)

hebasto commented:
cc @GBKS

GBKS commented:
That looks correctly centered now. Is this automatic centering or by setting pixel coordinates (or top padding)? If it's the latter, then it may not be centered anymore if the content on the left changes (like a translation being longer).

jarolrod commented:
@GBKS the centering is automatic, it's handled by the RowLayout. This change constrains the OptionSwitch to only the space needed for it's visual elements instead of having padding all around.

hebasto commented:
`root.width` == `indicator.implicitWidth` -- is it coincidence? If not, why a constant is duplicated?

jarolrod commented:
updated from 250bb9f to 5af9c73

changes: set indicators implicitWidth/height to parent width/height

shaavan commented on src/qml/controls/OptionSwitch.qml:
```diff
@@ -7,9 +7,12 @@ import QtQuick.Controls 2.15
 
 Switch {
     id: root
+    padding:0
```
nit:

```suggestion
    padding: 0
```

jarolrod replied:
done

shaavan reviewed (commented):
ACK modulo nit

- The optionswitch are correctly aligned.

Here's a screenshot I took on Ubuntu 22.04 with Qt version 5.15.3

**Screenshot:**
|Master|PR|
|---------|----|
|![Screenshot from 2022-09-07 00-36-42](https://user-images.githubusercontent.com/85434418/188718638-8004d7d9-9972-4a50-86c8-abd02c026582.png)|![Screenshot from 2022-09-07 00-28-21](https://user-images.githubusercontent.com/85434418/188718665-ff7b8da7-fda5-475f-8f4c-f8862c1cf502.png)|

jarolrod commented:
updated from [5af9c73 ](https://github.com/bitcoin-core/gui-qml/commit/5af9c73e46f28b5b71ef43c1ea8d269c18a82ff8)to [e6f8728](https://github.com/bitcoin-core/gui-qml/commit/e6f87282d39ee5a86aa2e1835280097e2843bb7e) ([pr161.01](https://github.com/jarolrod/gui-qml/tree/pr161.01) -> [pr161.02](https://github.com/jarolrod/gui-qml/tree/pr161.02), [compare](https://github.com/jarolrod/gui-qml/compare/pr161.01..pr161.02))

Changes: addressed spacing nit

hebasto commented:
https://github.com/bitcoin-core/gui-qml/pull/169#pullrequestreview-1101569629:

> Confirmed that this fixes the issue, and is a better approach than 161.
